apps/patient/src/views/Pharmacy.tsx

```diff
@@ -77,34 +147,77 @@ export const Pharmacy = () => {
   }
 
   const fetchPharmacies = async () => {
-    try {
-      graphQLClient.setHeader('x-photon-auth', token)
+    setLoadingMore(true)
 
-      const location = {
-        latitude,
-        longitude,
-        radius: 25
-      }
-      const limit = 3
-      const offset = pharmacyOptions.length
+    graphQLClient.setHeader('x-photon-auth', token)
 
-      const results: any = await graphQLClient.request(GET_PHARMACIES, { location, limit, offset })
+    const location = {
+      latitude,
+      longitude,
+      radius: 25
+    }
+    const limit = 3
+    const offset = pharmacyOptions.length
 
-      if (results) {
-        setPharmacyOptions([...pharmacyOptions, ...results.pharmaciesByLocation])
-      }
+    let results: any
+    try {
+      results = await graphQLClient.request(GET_PHARMACIES, { location, limit, offset })
     } catch (error) {
       console.error(JSON.stringify(error, undefined, 2))
       console.log(error)
 
-      if (error?.response?.errors) {
-        if (AUTH_HEADER_ERRORS.includes(error.response.errors[0].extensions.code)) {
-          navigate('/no-match')
-        } else {
-          setError(error.response.errors[0].message)
+      if (error?.response.errors[0].message === 'No pharmacies found near location') {
+        setShowingAllPharmacies(true)
+      }
+    }
+
+    if (results?.pharmaciesByLocation.length > 0) {
+      for (let i = 0; i < results.pharmaciesByLocation.length; i++) {
+        const name = results.pharmaciesByLocation[i].name
+        const address = results.pharmaciesByLocation[i].address
+          ? formatAddress(results.pharmaciesByLocation[i].address)
+          : ''
+
+        const placeRequest = {
+          query: name + ' ' + address,
+          fields: ['place_id']
+        }
+        const { response: place, status: placeStatus }: any = await query(
```

Comment:
are there any types for place and  placesStatus?

Reply:
I'll work on finding some. Wasn't a quick.